Formula/infer.rb

```diff
@@ -15,12 +15,21 @@ class Infer < Formula
 
   depends_on "autoconf" => :build
   depends_on "automake" => :build
+  depends_on "camlp4" => :build
   depends_on "cmake" => :build
+  depends_on "gnu-sed" => :build
   depends_on :java => ["1.7+", :build]
   depends_on "libtool" => :build
   depends_on "ocaml" => :build
   depends_on "opam" => :build
+  depends_on "perl" => :build
   depends_on "pkg-config" => :build
+  depends_on :x11 => :build
+  depends_on :xcode => :build
+  depends_on "gmp"
+  depends_on "mpfr"
+  depends_on "python@2"
+  depends_on "sqlite"
```

Comment:
Why (for these 5 deps: xcode, gmp, mpfr, python@2, sqlite)?

Reply:
- Full Xcode as a build dependency is required for the compilation of the custom clang infer uses. It provides certain headers that don't seem to be present in the commandline tools installation (discussion in [this infer issue](https://github.com/facebook/infer/issues/1002))
- `gmp`, `mpfr`, and `sqlite` provide dynamic libraries that are needed by infer at runtime (verified with `otool -L infer`)
- in the [earlier resolved conversation](https://github.com/Homebrew/homebrew-core/pull/40039#discussion_r285616176) between me and @jvillard in this PR he says python is required at runtime. it does not show up with `otool` but hopefully he can shed some light on that one?

Reply:
> * `gmp`, `mpfr`, and `sqlite` provide dynamic libraries that are needed by infer at runtime (verified with `otool -L infer`)

Does the build fail when they are removed? They may actually be dependencies of a different dependency used by infer, or configure step may be opportunistically finding and linking against them.

Although I just checked the documentation upstream and it appears they are needed.

> * Full Xcode as a build dependency is required for the compilation of the custom clang infer uses. It provides certain headers that don't seem to be present in the commandline tools installation (discussion in [this infer issue](https://github.com/facebook/infer/issues/1002))

The issue does not conclusively resolve that xcode is required. Did you try the `SDKROOT` recommendation?

Here too it appears upstream is requiring this, so it's probably OK.

Reply:
> does the build fail when they are removed

It does; i uninstalled each of them and ran tests; in all three cases the infer command failed out in ways that indicated missing dylibs.

> not conclusively resolve that xcode is required

Yup, tried the SDKROOT recommendation. I tested a clean build from source and a homebrew build, both on systems with full Xcode and without; the builds only complete with the full Xcode.